src/db/write.rs

```diff
@@ -338,15 +344,39 @@ impl<'a> Write<'a> {
 
     // Return value is the hash of the new commit.
     #[allow(clippy::too_many_arguments)]
-    pub async fn commit(mut self, head_name: &str) -> Result<String, CommitError> {
+    pub async fn commit(
+        mut self,
+        head_name: &str,
+        generate_diffs: bool,
+    ) -> Result<(String, BTreeMap<String, Vec<String>>), CommitError> {
```

Comment:
Instead of `BTreeMap<...>` for the changes, why not define a named type somewhere. That would give an opp to document the meaning of the key and the value (is it index name and changed keys?)

Reply:
+1

Reply:
ChangedKeysMap